wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/exception/ScmConfigurationPersistenceException.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.factory.server.scm.exception;
+
+/** Thrown when problem occurred during communication with permanent storage. */
+public class ScmConfigurationPersistenceException extends Exception {
```

Comment:
at this stage of factory component, we have no notion about where it could be store. It could be k8s secret or something else. So I made it an independent exception.